ecosystem/supersim.md

```diff
@@ -0,0 +1,319 @@
+
+Interop will usher in a paradigm shift in the way that developers build and deploy their applications, and a way to test their work locally is foundational to their success and thus the success of interop. 
+
+
+# High-Level Overview
+
+Supersim will allow developers to start multiple local Layer 2 chains with one command, and it will simulate the experience of interop via a command line interface. 
+
+# Proposed Solution
+
+Supersim will be a lightweight tool that simulates an interoperable Superchain environment locally. It will not require a complicated devnet setup and will be run using cli commands with configuration options that fall back to sensible defaults if they are not specified. Predefined state can be passed in at startup or the state can be forked based on specified block height.
+
+From a developer perspective, supersim will offer a command line interface through which to easily start up multiple local testnets that can be used interoperably via the [Optimism Interop Protocol](https://github.com/ethereum-optimism/specs/tree/main/specs/interop). 
+
+## Usage
+
+```bash
+// brings up two vanilla op chains with each other in the dep set with the same L1 chain
+
+supersim
+```
+
+```bash
+// brings up all forked chains in the superchain registry, if l1BlockHeight is not specified, uses the latest safe head
+
+supersim --superchain --l1BlockHeight=423012
+
+// can also specify a subset of members in the superchain
+
+supersim --superchain --networks=base,op-mainnet
+```
+
+config.toml(optional)
+
+```yaml
+[[networks]]
+forkRpcUrl = "https://sepolia.optimism.io"
+chainId = 420
+port = 8645
+dependencySet = []
+
+[[networks]]
+chainId = 99999
+port = 8646
+```
+
+## Architecture
+
+The **orchestrator** is responsible for spinning up the requested number of **anvil instances** and also for spinning up the **op-simulator**. It is also responsible for the interface between the **anvil instances** and the **op-simulator**.
+
+### Application Flow
+
+The flow of this tool will differ depending on whether the user opts for **vanilla** mode or **forking** mode.
+
+#### Vanilla mode
+
+    1. Orchestrator has preset chain configs and starts chains anew.
+    2. Orchestrator applies genesis state via the genesis-applier.
+    3. Orchestrator starts op-simulator and offers an interface between anvil instances and op-simulator.
+    4. Orchestrator monitors status of running local anvil instances.
+
+#### Forking mode 
+    
+    1. Orchestrator acquires L1 reference block to get the latest L2 height for forking using l2-fork-state-finder. 
+    2. Orchestrator gets appropriate chain configs from the superchain registry using the chain-config-loader.
+    3. Orchestrator starts the requested chains at determined L2 block height.
```

Comment:
Presumably this means that you'd need RPCs for all chains in your forked interop set, correct?

Reply:
For forking, yes I don't see a way around that. We were planning to default to public RPCs where possible and allow users to override.